lib/brain.js

```diff
@@ -522,15 +522,19 @@ Brain.prototype.scanPDF = function scanPDF () {
 }
 
 Brain.prototype.authorizeIdCardData = function () {
-  this.clearTimeoutToScanPDFCancel()
+  Promise.resolve()
+    .then(() => {
+      this.clearTimeoutToScanPDFCancel()
 
-  const customer = this.customer
-  const data = idCardData.getData()
-  this.trader.updateCustomer(customer.id, {idCardData: data})
```

Comment:
this was raising an error outside the promise context because customer were undefined.
the result was that the UI got stuck